### ISSUE TITLE
IO Error when installing httpfs extension from Python

### ISSUE THREAD

**Issue by davidgasquez** [duckdb/duckdb#3243]
#### What happens?
Executing `INSTALL 'httpfs';` returns the following error:
```
RuntimeError: IO Error: Failed to download extension http://extensions.duckdb.org/b52293246/linux_amd64/httpfs.duckdb_extension.gz
``` 

#### To Reproduce

First, install latest Python version of `duckdb`, then execute the following code:
```python
import duckdb
con = duckdb.connect(database=":memory:", read_only=False)
con.execute("INSTALL 'httpfs';")
```

#### Environment (please complete the following information):

I'm using the following dockerfile:

```Dockerfile
FROM mcr.microsoft.com/vscode/devcontainers/python:3.9
RUN pip3 install duckdb --pre --upgrade
ENTRYPOINT "/bin/bash"
``` 

An alternative way of reproducing the issue:
1. `docker run -it mcr.microsoft.com/vscode/devcontainers/python:3.9 /bin/bash`
2. `pip install duckdb`
3. `python -c 'import duckdb; con = duckdb.connect(database=":memory:", read_only=False); con.execute("INSTALL 'httpfs';");'`

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by davidgasquez** [comment]
Thanks for the context @samansmink! Let me know if I can test anything to help with this one!

**Comment by samansmink** [comment]
@gotdan the version you're using is from from before the https://github.com/duckdb/duckdb/pull/3245 fix and also from before we started uploading m1 binaries. https://github.com/duckdb/duckdb/pull/3201. Current master should be able to install all loadable extensions!

**Comment by gotdan** [comment]
Cool, thanks! What's the best way to download a build of the current master? If I do` pip install duckdb --upgrade --pre` (per https://duckdb.org/docs/installation/) the version that's installed still gives an error when I try to install json. Trying to download the macOS CLI binary listed on that page (https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_cli-osx-amd64.zip) returns a 404. Is there a different build page I should be looking at?

**Comment by samansmink** [comment]
@gotdan @davidgasquez this issue should be resolved now, could you let me know if it's working for you now?

**Comment by gotdan** [comment]
@samansmink - I haven't had a chance to try my own build yet, but pulling a fresh version from pypi is still giving me the IO error, and I'm still getting a 404 for the mac CLI download.

**Comment by davidgasquez** [comment]
Heya @samansmink! Probably related with https://github.com/duckdb/duckdb/issues/2804, I'm now seeing that error when reading remote files.

## Steps to reproduce

1. Run a shell inside a docker image: `docker run -it mcr.microsoft.com/vscode/devcontainers/python:3.9 /bin/bash`
2. Install latest DuckDB version: `pip install duckdb --upgrade --pre`
3. Install HTTPFS package: `python -c 'import duckdb; con = duckdb.connect(database=":memory:", read_only=False); con.execute("INSTALL 'httpfs';");'`
4. Execute any `select` to remote files. E.g: `con.execute("SELECT count(*) FROM 'https://raw.githubusercontent.com/owid/owid-datasets/master/datasets/CO2%20concentrations%20-%20NOAA%20(2019)/CO2%20concentrations%20-%20NOAA%20(2019).csv';")`

This is the full error message:

```
RuntimeError: IO Error: No files found that match the pattern "https://raw.githubusercontent.com/owid/owid-datasets/master/datasets/CO2%20concentrations%20-%20NOAA%20(2019)/CO2%20concentrations%20-%20NOAA%20(2019).csv"
```

**Comment by hannes** [comment]
@davidgasquez you need to `LOAD 'httpfs'` before trying to use it, too?

**Comment by davidgasquez** [comment]
Ups! Good one @hannes! Unfortunately, I'm getting another error:

```
>>> import duckdb
>>> con.execute("INSTALL 'httpfs';")
<duckdb.DuckDBPyConnection object at 0x7f382a66d6b0>
>>> con.execute("LOAD 'httpfs';")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: IO Error: File "/root/.duckdb/extensions/f1278198b/linux_amd64/httpfs.duckdb_extension" could not be loaded: /root/.duckdb/extensions/f1278198b/linux_amd64/httpfs.duckdb_extension: undefined symbol: _ZTIN6duckdb9ExceptionE
```

**Comment by samansmink** [comment]
@davidgasquez There does indeed seem to be a linking issue with the extensions that somehow doesn't occur during the current test. It probably means we need to set the RTLD_GLOBAL flag somewhere in the python package.

**Comment by davidgasquez** [comment]
Makes sense! Let me know if I can help with anything. [I'm now trying to replicate what goes on the Python GitHub Action Workflow](https://github.com/duckdb/duckdb/blob/master/.github/workflows/Python.yml) to see if the issue persist also when building from source.

**Comment by Mytherin** [comment]
As a temporary work-around for this, you can set the dlopen flags yourself before loading the duckdb module:

```py
import sys, os
sys.setdlopenflags(os.RTLD_GLOBAL | os.RTLD_NOW)

import duckdb
con = duckdb.connect()
con.execute('load httpfs')

```

**Comment by davidgasquez** [comment]
Tried that and seems the `undefined symbol` error is still there. 

```
❯ docker run -it mcr.microsoft.com/vscode/devcontainers/python:3.9 /bin/bash
root ➜ / $ pip install duckdb --upgrade --pre
root ➜ / $ python
Python 3.9.10 (main, Feb  8 2022, 05:06:38)
[GCC 10.2.1 20210110] on linux
>>> import sys, os
>>> sys.setdlopenflags(os.RTLD_GLOBAL | os.RTLD_NOW)
>>> import duckdb
>>> con = duckdb.connect()
>>> con.execute('install httpfs')
<duckdb.DuckDBPyConnection object at 0x7f994361c8b0>
>>> con.execute('load httpfs')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: IO Error: File "/root/.duckdb/extensions/f1278198b/linux_amd64/httpfs.duckdb_extension" could not be loaded: /root/.duckdb/extensions/f1278198b/linux_amd64/httpfs.duckdb_extension: undefined symbol: _ZN6duckdb10FileSystem15DirectoryExistsERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
```

**Comment by samansmink** [comment]
@davidgasquez Yea this is actually another issue I believe. Somehow the duckdb binary for the python wheel seems to get compiled with GLIBCXX_USE_CXX11_ABI=0 ([url](https://gcc.gnu.org/onlinedocs/gcc-5.2.0/libstdc++/manual/manual/using_dual_abi.html)). Strangely, when i compile the duckdb python package manually the correct CX11 symbols are actually generated, so somehow there's either an environment difference or the way setup.py is called is different.

**Comment by davidgasquez** [comment]
I see! Are there any workarounds to avoid the issue?

**Comment by samansmink** [comment]
we're working on a solution that should be up for PR soon, we're adding binary builds with the GLIBCXX_USE_CXX11_ABI=0 flag for python
